### PR TITLE
Deploy CLI Tool new documentation

### PR DESCRIPTION
**Pull request by tamigoodall**
Added new files

### CODE REVIEW

File: articles/extensions/deploy-cli/guides/import-export-directory-structure.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+   ```json
+   {
+     "AUTH0_DOMAIN": "<YOUR_TENANT>.auth0.com",
+     "AUTH0_CLIENT_ID": "<client_id>",
+     "AUTH0_CLIENT_SECRET": "<client_secret>",
+     "AUTH0_KEYWORD_REPLACE_MAPPINGS": {
+       "AUTH0_TENANT_NAME": "<NAME>",
+       "ENV": "DEV"
+   },
```
Comment (fyockm) [inline articles/extensions/deploy-cli/guides/import-export-directory-structure.md]: spacing is off for this brace. should be bumped 2 spaces to the right.

File: articles/extensions/deploy-cli/guides/install-deploy-cli.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  * delete:grants
+  * read:guardian_factors
+  * update:guardian_factors
+  * read:email_templates
+  * create:email_templates
+  * update:email_templates
+
+Use the [Auth0 Deploy CLI Extension](https://github.com/auth0-extensions/auth0-deploy-cli-extension/blob/master/README.md) to create the application. 
+
+## Install the tool using `bash`
```
Comment (fyockm) [inline articles/extensions/deploy-cli/guides/install-deploy-cli.md]: This sounds a little strange to me. I think it should read "Install the tool using the command line", or something like that.

File: articles/extensions/deploy-cli/guides/install-deploy-cli.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+```
+bash
```
Comment (fyockm) [inline articles/extensions/deploy-cli/guides/install-deploy-cli.md]: `bash` should be on the previous line (with the ``` marks)

File: articles/extensions/deploy-cli/guides/test-locally.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+contentType:
+  - how-to
+useCase: extensibility-extensions
+---
+# Test Deploy CLI Operations Locally
+
+Clone the GitHub repo and install globally.
+
+```
+bash
```
Comment (fyockm) [inline articles/extensions/deploy-cli/guides/test-locally.md]: as above, i think "bash" should be on the previous line _with_ the tick marks ```bash

File: articles/extensions/deploy-cli/guides/import-export-directory-structure.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+   Use the `client ID` and secret from your newly created client (the client is named `auth0-deploy-cli-extension` if you used the extension).
+
+   By default the tool merges your current environment variables and overrides `config.json` which has the same top key. Use the `--no-env` option to disable the override via the command line.
+
+   You can either set env variables or place the values in a config file anywhere on the file system.
+
+2. Run deploy.
+
+   ```
+   bash
```
Comment (fyockm) [inline articles/extensions/deploy-cli/guides/import-export-directory-structure.md]: `bash` should be on the previous line (with the ``` marks)

File: articles/extensions/deploy-cli/guides/install-deploy-cli.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  * update:tenant_settings
+  * read:grants
+  * delete:grants
+  * read:guardian_factors
+  * update:guardian_factors
+  * read:email_templates
+  * create:email_templates
+  * update:email_templates
+
+Use the [Auth0 Deploy CLI Extension](https://github.com/auth0-extensions/auth0-deploy-cli-extension/blob/master/README.md) to create the application.
```
Comment (fyockm) [inline articles/extensions/deploy-cli/guides/install-deploy-cli.md]: Let's put a header above this. "Install using the Auth0 Deploy CLI Extension" or similar.

File: articles/extensions/deploy-cli/references/environment-variables-keyword-mappings.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  - If you use the `@` symbols, it will do a `JSON.stringify` on your value before replacing it.  So if it is a string, it will add quotes. and if it is an array or object, it will add braces.  
+
+  - If you use the `#` symbol instead, it will just do a literal replacement; it will not add quotes or brackets.
+
+::: note
+By default the tool will also merge your current environment variables and override the **AUTH0_KEYWORD_REPLACE_MAPPINGS** which have the same top key. You can disable this via the command line with the `--no-env` option.
+:::
+
+For example, you could specify a different JWT timeout in your dev environment then prod for testing and a different environment URL.
```
Comment (fyockm) [inline articles/extensions/deploy-cli/references/environment-variables-keyword-mappings.md]: s/then/than

File: articles/extensions/deploy-cli/guides/test-locally.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+---
+title: Test Deploy CLI Operations Locally
```
Comment (fyockm) [inline articles/extensions/deploy-cli/guides/test-locally.md]: to be honest, I'm not sure that this actually belongs in the docs. it's fine on the repo readme, but seems out of place here.

File: articles/extensions/deploy-cli/references/deploy-cli-options.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+## Commands
+
+- `a0deploy import` Deploy Configuration
+- `a0deploy export` Export Auth0 Tenant Configuration
+
+## Options
+- `--help` Show help  [boolean]
+- `--version` Show version number  [boolean]
+- `--verbose, -v` Dump extra debug information.  `[string] [default: false]`
```
Comment (fyockm) [inline articles/extensions/deploy-cli/references/deploy-cli-options.md:24]: inconsistent formatting. "[boolean]" above is not monospace, but `[string] [default: false]` and `[string]` on the line below are

File: articles/extensions/deploy-cli/index.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+Auth0 supports continuous integration and deployment (CI/CD) of Auth0 tenants through our [source control extensions](/extensions#deploy-hosted-pages-rules-and-database-connections-scripts-from-external-repositories) and integration into existing CI/CD pipelines by using the Deploy CLI tool.
+
+The Deploy CLI tool (`auth0-deploy-cli`) supports two methods to import and export the following Auth0 tenant configuration objects: 
+
+- Tenant Settings
+- Rules (Including Secrets/Settings)
+- Connections
+- Custom Databases
+- Clients / Applications
+- Resource Servers (API's)
```
Comment (fyockm) [inline articles/extensions/deploy-cli/index.md]: should be `APIs` without the apostrophe

**Review by fyockm — CHANGES_REQUESTED** [review]

File: articles/extensions/deploy-cli/guides/test-locally.md (hunk 8 above)
Comment (tamigoodall) [reply articles/extensions/deploy-cli/guides/test-locally.md]: I removed this file and all the links to it.

File: articles/extensions/deploy-cli/guides/test-locally.md (hunk 4 above)
Comment (tamigoodall) [reply articles/extensions/deploy-cli/guides/test-locally.md]: File removed.

File: articles/extensions/deploy-cli/references/environment-variables-keyword-mappings.md (hunk 7 above)
Comment (tamigoodall) [reply articles/extensions/deploy-cli/references/environment-variables-keyword-mappings.md]: Actually, the sentence should say:

> For example, you could specify a different JWT timeout in your dev environment,  **and** then **use** prod for testing and a different environment URL.

**Review by fyockm — DISMISSED** [review]

**Comment by erjohnson** [comment]
Should add an entry to https://auth0.com/docs/extensions for this. Also that page could use some reformatting but that can be done in a later PR

**Comment by tamigoodall** [comment]
> Should add an entry to https://auth0.com/docs/extensions for this. Also that page could use some reformatting but that can be done in a later PR

It is linked on the GitHub Deployment link on that page. The requester didn't really want it on the Extensions page directly, because he believes it's not really an extension but a tool. I'm not sure I agree. From a user's perspective, how would they know the difference. And you can add it using the Extensions tab from the dashboard, so...?

**Review by erjohnson — APPROVED** [review]

**Pull request merged**